core/src/main/java/hudson/PluginManager.java

```diff
@@ -1966,6 +1966,13 @@ public HttpResponse doUploadPlugin(StaplerRequest req) throws IOException, Servl
             return FormValidation.error(Messages.PluginManager_emptyUpdateSiteUrl());
         }
 
+        //Check to see if the value is a file as some update centers are local json files
```

Comment:
will involve a file lookup for something like `http://foo.example.com/bar` 
e.g.
 `new File("http://foo.example.com/bar").exists()`
 
 Rather than check for something that is obviously bogus perhaps first the URI should be created.
 Then if it is a file based one do X, if it is http/https do Y (create another URI from the parsed URI appending the query parameter for the version)
 if it is anything else return an `FormValidation.warning("unable to check validity of update center due to protocol -> the protocol")`